src/constructors.jl

```diff
@@ -0,0 +1,147 @@
+abstract type AbstractHMCSampler <: AbstractMCMC.AbstractSampler end
+
+##############
+### Custom ###
+##############
+
+"""
+    HMCSampler
+
+A `AbstractMCMC.AbstractSampler` for kernels in AdvancedHMC.jl.
+
+# Fields
+
+$(FIELDS)
+
+# Notes
+
+Note that all the fields have the prefix `initial_` to indicate
+that these will not necessarily correspond to the `kernel`, `metric`,
+and `adaptor` after sampling.
+
+To access the updated fields use the resulting [`HMCState`](@ref).
+"""
+Base.@kwdef struct HMCSampler{
+    K<:AbstractMCMCKernel,
+    M<:AbstractMetric,
+    A<:Adaptation.AbstractAdaptor,
+} <: AbstractHMCSampler
+    "[`AbstractMCMCKernel`](@ref)."
+    kernel::K
+    "[`AbstractMetric`](@ref)."
+    metric::M
+    "[`AbstractAdaptor`](@ref)."
+    adaptor::A
+    "Adaptation steps if any"
+    n_adapts::Int = 0
+end
+
+############
+### NUTS ###
+############
+"""
+    NUTS(n_adapts::Int, δ::Float64; max_depth::Int=10, Δ_max::Float64=1000.0, init_ϵ::Float64=0.0)
+
+No-U-Turn Sampler (NUTS) sampler.
+
+Usage:
+
+```julia
+NUTS()            # Use default NUTS configuration.
+NUTS(1000, 0.65)  # Use 1000 adaption steps, and target accept ratio 0.65.
+```
+
+Arguments:
+
+- `n_adapts::Int` : The number of samples to use with adaptation.
```

Comment:
Generally speaking, I'd recommend just moving these docstrings to the fields themselves and use the $(FIELDS) to interpolate these directly into the docstring, rather than keeping track of this manually (see how it's done for `HMCSampler`)